src/core.ts

```diff
@@ -355,36 +355,63 @@ export type TrackingConsent = {
   [purpose: string]: boolean | undefined;
 };
 
-/** Tracking purpose consent config with timestamp & auth metadata */
-export type TrackingConsentDetails = {
-  /** Tracking consent config */
-  purposes: TrackingConsent;
-  /**
-   * Was tracking consent confirmed by the user?
-   * If this is false, the consent was resolved from defaults & is not yet confirmed
-   */
-  confirmed: boolean;
-  /** Consent resolution/last-modified timestamp (ISO 8601) */
-  timestamp: string;
-  /** Has the consent been updated (including no-change confirmation) since default resolution */
-  updated?: boolean;
-  /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
-  prompted?: boolean;
-  /** Transparency Consent (TCF) String */
-  tcf?: string;
-  /** US Privacy (USP) String */
-  usp?: string;
-  /** Global Privacy Platform (GPP) String */
-  gpp?: string;
-  /** Consent Manager View State */
-  viewState?: ViewState;
-  /** Airgap Version */
-  airgapVersion?: string;
-  /** Arbitrary metadata that customers want to be associated with consent state */
-  metadata?: unknown;
-  /** When the metadata was last updated */
-  metadataTimestamp?: string;
-};
+export const TrackingConsentDetails = t.intersection([
+  t.type({
+    /** Tracking consent config */
+    purposes: t.record(t.string, t.boolean),
+    /**
+     * Was tracking consent confirmed by the user?
+     * If this is false, the consent was resolved from defaults & is not yet confirmed
+     */
+    confirmed: t.boolean,
+    /** Consent resolution/last-modified timestamp (ISO 8601) */
+    timestamp: t.string,
+  }),
+  t.partial({
+    /** Has the consent been updated (including no-change confirmation) since default resolution */
+    updated: t.boolean,
+    /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
+    prompted: t.boolean,
+    /** Transparency Consent (TCF) String */
+    tcf: t.string,
+    /** US Privacy (USP) String */
+    usp: t.string,
+    /** Global Privacy Platform (GPP) String */
+    gpp: t.string,
+    /** Consent Manager View State */
+    viewState: valuesOf(ViewState),
+    /** Airgap Version */
+    airgapVersion: t.string,
+    /** Arbitrary metadata that customers want to be associated with consent state */
+    metadata: t.unknown,
+    /** When the metadata was last updated */
+    metadataTimestamp: t.string,
+  }),
+]);
+
+/** Override types. */
+export type TrackingConsentDetails = t.TypeOf<typeof TrackingConsentDetails>;
+
+export const ConsentPreferencesBody = t.type({
+  token: t.string,
+  partition: t.string,
```

Comment:
This is changed from [sombra-types](https://github.com/transcend-io/main/blob/8f0e8fffc67410017a0add1575fea8349037ca7a/frontend-support/sombra-types/src/codecs.ts#L1222-L1229) where consentManagerId is part of ConsentPreferencesBody and partition is optional.

Following huddle w/ @anotherminh and @eligrey, partition will now be required and consentManagerId will not be included. (In the event where partition is undefined, it can fall back to consentManagerId.)

Reply:
cool, thanks!!